### ISSUE TITLE
handling imported vasp jobs for Phonopy calculations 

### ISSUE THREAD

**Issue by sinazadeh** [pyiron/pyiron#1378]
I am trying to figure out how to do Phonopy calculations using pyiron+vasp. I tried this code but it raises 'JobPath' object has no attribute 'create_job' error. I will appreciate it if anyone can explain to me what am I doing wrong.

**WHAT I WANT TO DO**
Calculate thermodynamic properties using VASP + Pyiron. If there is easier way to do so, I would appreciate if you share with me.

```
%matplotlib inline
import matplotlib.pyplot as plt
import numpy as np
from pyiron import Project
import pandas as pd

project_folder='phononpy_proj'
pr = Project(path=project_folder)
basis = pr.create_ase_bulk('Al', cubic=True)

for temperature in np.linspace(200, 1200, 5):
    job = pr.create_job(pr.job_type.Vasp,'Al_T{}K'.format(int(temperature)))
    job.structure = basis
    job.calc_md(temperature=temperature)
    job.server.queue = job.server.list_queues()[-1]
    job.run()
```

And

```
import numpy as np
import matplotlib.pyplot as plt
from pyiron import Project

vasp_project_folder = 'phononpy_proj'

pr = Project('imported_phononpy_proj')
pr.import_from_path(path=vasp_project_folder, recursive=True)    

for job in pr.iter_jobs(convert_to_object=False):
    phono = job.create_job(pr.job_type.PhonopyJob, "phono")
    phono.run()
```

**Comment by sinazadeh** [comment]
changing 

`for job in pr.iter_jobs(convert_to_object=False):`
to
`for job in pr.iter_jobs(convert_to_object=True):`
raises:
ValueError: ('GenericMaster requires reference jobs to have status initialized, rather than ', 'finished')

**Comment by jan-janssen** [comment]
Hi @sinazadeh, 

The syntax is a bit different: 
```
job = pr.create_job(
    job_type=pr.job_type.Vasp,
    job_name='Al_T{}K'.format(int(temperature))
)
job.structure = basis
job.server.queue = job.server.list_queues()[-1]
phono = job.create_job(
    job_type=pr.job_type.PhonopyJob,
    job_name='phono_Al_T{}K'.format(int(temperature))
)
phono.run()
```
The idea behind this is that phonopy is used to create the different displacements based on the VASP template you defined before. So you can configure the energy cut-off, kpoint mesh and so one once, and phonopy is just changing the structure. 

In addition wed have an example how to combine phonopy and LAMMPS: 
https://github.com/pyiron/pyiron/blob/master/notebooks/phonopy_example.ipynb
And a tutorial as part of the virtual course:
https://pyiron.org/phasediagram-workshop-2020/ThermodynamicProperties.html

**Comment by sinazadeh** [comment]
> Hi @sinazadeh,
> 
> The syntax is a bit different:
> 
> ```
> job = pr.create_job(
>     job_type=pr.job_type.Vasp,
>     job_name='Al_T{}K'.format(int(temperature))
> )
> job.structure = basis
> job.server.queue = job.server.list_queues()[-1]
> phono = job.create_job(
>     job_type=pr.job_type.PhonopyJob,
>     job_name='phono_Al_T{}K'.format(int(temperature))
> )
> phono.run()
> ```
> 
> The idea behind this is that phonopy is used to create the different displacements based on the VASP template you defined before. So you can configure the energy cut-off, kpoint mesh and so one once, and phonopy is just changing the structure.
> 
> In addition wed have an example how to combine phonopy and LAMMPS: https://github.com/pyiron/pyiron/blob/master/notebooks/phonopy_example.ipynb And a tutorial as part of the virtual course: https://pyiron.org/phasediagram-workshop-2020/ThermodynamicProperties.html

Thank you for your help. I have a follow-up question. Is it possible to plot vasp results similar to the way we do for LAMMPS?
For example something like this:
```
import numpy as np
import matplotlib.pyplot as plt
from pyiron import Project

vasp_project_folder = 'phononpy_proj'

pr = Project('imported_phononpy_proj')
pr.import_from_path(path=vasp_project_folder, recursive=True)    

for job in pr.iter_jobs(convert_to_object=True):
    temperatures = np.linspace(0,800,41)
    thermo = phono.get_thermal_properties(temperatures=temperatures)
    phono.plot_dos()
```

**Comment by jan-janssen** [comment]
The confusing part for me is that you import the calculation rather than using pyiron to execute them.

**Comment by sinazadeh** [comment]
I am not sure how to do that because in the system that I am working on, we need to submit vasp jobs in queue. After calculations I need to import them, right? Or I am missing something.

**Comment by jan-janssen** [comment]
No, you should be able to just execute the notebook again and pyiron can load the existing calculation. If you use the example code I posted above it is submitting the calculation to the queue and once the calculation are finished the phonopy object is update, so the next time you execute it you can plot the results.

**Comment by sinazadeh** [comment]
> No, you should be able to just execute the notebook again and pyiron can load the existing calculation. If you use the example code I posted above it is submitting the calculation to the queue and once the calculation are finished the phonopy object is update, so the next time you execute it you can plot the results.

Thank you for your time and sorry for my late response. I was studying the links you sent to me. I copied the code you wrote but still, there is an error. The following is the code:

```
%matplotlib inline
import matplotlib.pyplot as plt
import numpy as np
from pyiron import Project
import pandas as pd

project_folder='phononpy_proj2'
pr = Project(path=project_folder)
basis = pr.create_ase_bulk('Fe', cubic=True)

for temperature in np.linspace(0, 800, 5):
    job = pr.create_job(
        job_type=pr.job_type.Vasp,
        job_name='Al_T{}K'.format(int(temperature))
    )
    job.structure = basis
    job.server.queue = job.server.list_queues()[-1]
    phono = job.create_job(
        job_type=pr.job_type.PhonopyJob,
        job_name='phono_Al_T{}K'.format(int(temperature)))
    phono.run()

thermo = phono.get_thermal_properties(temperatures=np.linspace(0, 800, 5))
plt.plot(temperatures, thermo.cv)
plt.xlabel("Temperatures")
plt.ylabel("Heat Capacity")
```

It raises the following error:
`RuntimeError: run_mesh has to be done beforerun_thermal_properties.`

**Comment by jan-janssen** [comment]
After the `run()` command the calculations are submitted to the queue, so most likely you reach the `get_thermal_properties()` part before they are finished. If you wait inside your script you can use `pr.wait_for_jobs()`. The alternative is to execute the script again after all calculations are finished, then the plotting should work just fine. To check the status of your calculation you can use a script like: 

```
from pyiron import Project
pr = Project(path='phononpy_proj2')
print(pr.job_table()
```

When you execute the script again rather than executing the calculation again pyiron reloads them from the database and plots the results. 

In addition I noticed that you do not change anything in dependence of the temperature other then the job name - so it is basically five times the same calculation. I assume you want to adjust the volume according to the temperature to consider the quasi-harmonic approximation.

**Comment by sinazadeh** [comment]
> After the `run()` command the calculations are submitted to the queue, so most likely you reach the `get_thermal_properties()` part before they are finished. If you wait inside your script you can use `pr.wait_for_jobs()`. The alternative is to execute the script again after all calculations are finished, then the plotting should work just fine. To check the status of your calculation you can use a script like:
> 
> ```
> from pyiron import Project
> pr = Project(path='phononpy_proj2')
> print(pr.job_table()
> ```
> 
> When you execute the script again rather than executing the calculation again pyiron reloads them from the database and plots the results.
> 
> In addition I noticed that you do not change anything in dependence of the temperature other then the job name - so it is basically five times the same calculation. I assume you want to adjust the volume according to the temperature to consider the quasi-harmonic approximation.

I waited enough time to finish the calculations. Also, if I import the folder I can see the outputs. Is it possible that it has something to do with the way that I configured files in VASP and Queues folder in the pyiron resources folder?

**Comment by jan-janssen** [comment]
So in the `job_table` all calculation are listed as finished? if that is the case then there is no issue with the resource folder, if they are listed as `submitted` or `aborted` it is most likely an issue with the resource folder.

**Comment by sinazadeh** [comment]
> So in the `job_table` all calculation are listed as finished? if that is the case then there is no issue with the resource folder, if they are listed as `submitted` or `aborted` it is most likely an issue with the resource folder.

Yes, there is a problem with the job submition. I open a new issue for that. Thank you.
